index.bs

```diff
@@ -6714,6 +6735,52 @@ During a transition from the FIDO U2F JavaScript API, a [=[RP]=] may have a popu
 : Authenticator extension output
 :: None.
 
+### Conditional Create Extension (<dfn>conditionalCreate</dfn>) ### {#sctn-authenticator-conditional-create-extension}
+
+This [=client extension|client=] [=authentication extension=] indicates that the [=[RP]=] would like to create a credential if the authentication ceremony is completed successfully.
+
+
+: Extension identifier
+:: `conditionalCreate`
+
+: Operation applicability
+:: [=authentication extension|Authentication=]
+
+: Client extension input
+:: The Boolean value [TRUE] to indicate that this extension is requested by the [=[RP]=].
+    <xmp class="idl">
+    partial dictionary AuthenticationExtensionsClientInputs {
+        boolean conditionalCreate;
+    };
+    </xmp>
+
+: Client extension processing
+
+:: When [[#sctn-getAssertion|assertion]]:
+        1. Just after [establishing the RP ID](#GetAssn-DetermineConditional) perform these steps:
+
+            1. If <code>|options|.{{CredentialRequestOptions/mediation}}</code> is not present or does not have the value
+                {{CredentialMediationRequirement/conditional}}:
+
+                1. Throw a "{{NotAllowedError}}" {{DOMException}}
+
+        1. When [showing conditional assertion form control](#GetAssn-ConditionalMediation-Interact-FormControl):
+
+            1. Inform the user that the [=[RP]=] will create a credential if the user agent mediates the authentication ceremony.
+
+            1. Set |conditionalCreateLifetimeTimer| to a client-specific default.
+
+            1. Set |conditionalCreateOrigin| to the current origin.
+
+        1. If the user agent mediates the authentication ceremony while the conditional assertion is running:
+
+            1. Start |conditionalCreateLifetimeTimer|.
+
+            1. Optionally collect an [=authorization gesture=] confirming [=user consent=] to create the credential which may later be used to skip (this step)[#op-makecred-step-user-consent].
```

Comment:
In this PR's current incarnation, would an RP that **cannot** do conditional UI (e.g. the RP requires username submission to check that the user can use WebAuthn according to access policy) still be able to conditionally register a passkey? I suppose I wouldn't want to see this functionality require the user to have to click something from the browser's autofill UI.

In Cisco's case, we would want to be able to do the following...

```js
/**
 * No form is actually shown at this point, we're just making this call to allow for
 * the subsequent .create() to silently register
 */
navigator.credentials.get({
  mediation: 'conditional',
  publicKey: { ..., extensions: { conditionalCreate: true } },
});
/**
 * Wait a short period of time since the .get() above is asynchronous and we want to
 * give the browser time to start the `conditionalCreateLifetimeTimer`
 */
setTimeout(() => {
  navigator.credentials.create({
    mediation: 'conditional',
    publicKey: { ... },
  }).then(sendRegRespToServer);
}, 100);
```

...so that we too would be able to use this capability to reduce friction when onboarding users to passkeys. But the way I'm reading this we would **not** be able to leverage conditional creation because the user didn't click the browser's autofill.

Can we consider ways to accommodate a use case like ours too? I'm not 100% sure why autofill interaction is required for this to be pulled off.

Reply:
Yes. The extension is not required.

Reply:
Ah, I missed this recent change to the PR:

![Screenshot 2023-09-20 at 12 19 54 PM](https://github.com/w3c/webauthn/assets/5166470/95109edf-b39c-4e9e-9957-a64dc7c0c6ae)

So then an RP that can't do conditional UI can simply do the conditional `.create()` call after a successful auth (assuming they check that the conditional create capability is present)...

```
navigator.credentials.create({
    mediation: 'conditional',
    publicKey: { ... },
  }).then(sendRegRespToServer);
```

...but would be at the mercy of the browser to have **implicitly** started `conditionalCreateLifetimeTimer` after the previous modal `.get()`? That would mean "conditional UI + the `conditionalCreate` extension" is an **explicit** way to ensure `conditionalCreateLifetimeTimer` gets started?

Having thought this through a bit, perhaps we need a way for the RP to query if `conditionalCreateLifetimeTimer` is not expired. Or perhaps this could get factored into whether a "client-create" capability from `getClientCapabilities()` returns true.

Reply:
Taking it as an action item to add a recommended minimum time here.

Reply:
From the WG meeting on 9/20: I think now what's _actually_ needed is a suggested duration for `conditionalCreateLifetimeTimer` that encourages user agents to give RP's a similar amount of time to make the conditional create call.